nexus/db-queries/src/db/datastore/ip_pool.rs

```diff
@@ -587,23 +588,40 @@ impl DataStore {
 
         let conn = self.pool_connection_authorized(opctx).await?;
 
+        // We always insert and update the record on conflicts.
```

Comment:
In the public-facing API, this is a POST request - normally, this wouldn't be an idempotent operation, which makes it odd to have this be doing `on_update` instead of throwing a conflict-related error.

Is this change actually necessary for the constraints to apply? Won't we still see UniqueViolation constraints even if we aren't doing "on_conflict do update"?

Reply:
Fixed in 8d3ea334e68ad75b6db19975a1dbc9fae6d0d5b5 and c982a599326225be10ebfe3db9707ec93819b67c